### DISCUSSION TITLE
Import Produt Batch Saving + Event Bus [Gets only Latest Event]

### DISCUSSION THREAD

**Discussion by Yuri-Lima** [vendurehq/vendure#2468] [General]
I had an interesting case where, i was saving only 5 products to DB and each  save transicional gets done, i sent an event bus to create a new event on my TimeLine [Activity Stream].

However because of the `.pipe(debounceTime(50))` only the latest event was showing.
After a investigating the case with my manger, we came across of that solution, instead of use `.pipe(debounceTime(50))` we used ``.pipe(delay(50))`` from `import { delay } from "rxjs/operators";`, which solve our case perfectly.

I am wondering if the Vendure has the same case, of corse if its uses TimeLine!!

Graph Example of the difference: https://reactive.how/debouncetime

**Comment by michaelbromley** [comment]
Hi, thanks for sharing! Yes, `debounceTime` will drop events until the end of the time window, that's something to look out for!

We do use `debounceTime` in 2 locations in Vendure core, but in those cases we don't suffer from any loss of data, as we just use the events as a trigger for some logic which fetches fresh data from the DB.
